src/Sulu/Bundle/SnippetBundle/Content/SnippetDataProvider.php

```diff
@@ -67,20 +72,34 @@ class SnippetDataProvider implements DataProviderInterface
      */
     private $referenceStore;
 
+    /**
+     * @var FormMetadataProvider|null
+     */
+    private $formMetadataProvider;
+
+    /**
+     * @var TokenStorageInterface|null
+     */
+    private $tokenStorage;
+
     public function __construct(
         ContentQueryExecutorInterface $contentQueryExecutor,
         ContentQueryBuilderInterface $snippetQueryBuilder,
         SuluNodeHelper $nodeHelper,
         LazyLoadingValueHolderFactory $proxyFactory,
         DocumentManagerInterface $documentManager,
-        ReferenceStoreInterface $referenceStore
+        ReferenceStoreInterface $referenceStore,
+        FormMetadataProvider $formMetadataProvider = null,
+        TokenStorageInterface $tokenStorage = null
     ) {
         $this->contentQueryExecutor = $contentQueryExecutor;
         $this->snippetQueryBuilder = $snippetQueryBuilder;
         $this->nodeHelper = $nodeHelper;
         $this->proxyFactory = $proxyFactory;
         $this->documentManager = $documentManager;
         $this->referenceStore = $referenceStore;
+        $this->formMetadataProvider = $formMetadataProvider;
```

Comment:
The `tokenStorage` probably is really nullable, but the `formMetadataProvider` should trigger a deprecation notice here if not set, shouldn't it?